### ISSUE TITLE
Sewer Pub

### ISSUE THREAD

**Issue by Draylontis** [StardewModders/mod-ideas#547]
where monsters, the talking animals, wizard and others can come for a drink.

**Comment by lemurkat** [comment]
Waiting for NPC tricks to create this one! Need to be able to warp Krobus into it.

**Comment by FastestestBoi** [comment]
it sounds amazing

**Comment by SwiftIsHere** [comment]
Attached content pack lets all NPCs use the sewer grate in the town square, with an option to turn on the forest grate as well. Note that a) the player will be able to use the warp as well, and b) NPCs appear to always prefer the town grate when both are active, and I can't for the life of me figure out how to change that. Please feel free to take and play with the code if you like! 
[[CP] Unlock the Sewer.zip](https://github.com/StardewModders/mod-ideas/files/5199249/CP.Unlock.the.Sewer.zip)

**Comment by lemurkat** [comment]
If you turn both on at the same rime you will activate a circular route and may end up with NPCs wandering through the sewers to get frok forest to town.

**Comment by SwiftIsHere** [comment]
Yep, that's another issue – why I disabled the forest warp by default. Though given how reluctant they are to walk through the forest warp (which I think might be related to the sewer leak tiles not being having a type? Though nothing seemed to change when I made them grass) I'm not sure if it'll be an actual problem.

**Comment by lemurkat** [comment]
Okay created the pub itself but struck a problem when I tried to move Dwarf into it. Krobus and dwarf have non standardized sprite sheets and it keeps grabbing the 32 pixel high instead of their 24.

Also no NPC wants to use the doors so I'll need to find an alternative access route.

**Comment by Pathoschild** [comment]
@lemurkat [Standardized Krobus](https://www.moddrop.com/stardew-valley/mods/716314-standardized-krobus) might help. It doesn't handle the Dwarf yet, but it's [open-source if you want to peek at the code](https://github.com/minervamaga/StandardizedKrobus).
